extensions/ihosts/src/utils/file.ts

```diff
@@ -0,0 +1,94 @@
+import { getPreferenceValues } from "@raycast/api";
+import crypto from "node:crypto";
+import fs from "node:fs";
+import os from "node:os";
+import path from "node:path";
+import { runAppleScript } from "run-applescript";
+import { SystemHostFilePath, TempSystemHostFileName } from "../const";
+
+export function getSysHostFileHash(): string {
+  const sysHostFileBuf = fs.readFileSync(SystemHostFilePath);
+  return crypto.createHash("md5").update(sysHostFileBuf).digest("hex");
+}
+
+export function getSysHostFile(): string {
+  return fs.readFileSync(SystemHostFilePath, "utf8");
+}
+
+export function getContentFromFile(path: string): string {
+  return fs.readFileSync(path, "utf8");
+}
+
+export async function writeSysHostFile(content: string) {
+  const tempHostsPath = path.join(os.tmpdir(), TempSystemHostFileName);
+  fs.writeFileSync(tempHostsPath, content, "utf8");
+  checkSysHostAccess() || (await getSysHostAccess());
+  await runAppleScript(`do shell script "cat ${tempHostsPath} > ${SystemHostFilePath} && rm -rf ${tempHostsPath}"`);
```

Comment:
any reason to use an applescript instead of a child process directly?

Reply:
Indeed, there is no need to use apple script～ I have pushed a commit to make the change